playbooks/prepare_stack.yaml

```diff
@@ -111,6 +111,17 @@
     environment:
       OS_CLOUD: standalone
 
+  - name: Create Manila share type for CephFS NFS # noqa 301
+    shell: |
+      if ! openstack share type show cephfsnfstype; then
+          openstack share type create cephfsnfstype false
```

Comment:
In my experience, this is not enough, I had to explicitly disable snapshot_support for the share type otherwise manila scheduler would fail to find a suitable host:

```
standalone ❯ openstack share type show cephfsnfstype                                                                                                                      
+----------------------+--------------------------------------+                                                                                                           
| Field                | Value                                |                                                                                                           
+----------------------+--------------------------------------+                                                                                                           
| id                   | 7d662e40-40e1-4282-915e-d4198d064a4a |                                                                                                           
| name                 | cephfsnfstype                        |                                                                                                           
| visibility           | public                               |                                                                                                           
| is_default           | False                                |                                                                                                           
| required_extra_specs | driver_handles_share_servers : False |                                                                                                           
| optional_extra_specs | snapshot_support : True              |                                                                                                           
| description          | None                                 |                                                                                                           
+----------------------+--------------------------------------+                                                                                                           
```

```
2021-05-19 06:43:17.848 7 DEBUG manila.scheduler.filters.base [req-296dc84b-c365-46d5-a3b0-c665420b70af 076decc351c94a3487594883f7420966 1b1e4fae15e045eaa66f8ee2c38e06ea - - -] Filter CapacityFilter returned 1 host(s) get_filtered_objects /usr/lib/python3.6/site-packages/manila/scheduler/filters/base.py:97
2021-05-19 06:43:17.848 7 DEBUG manila.scheduler.utils [req-296dc84b-c365-46d5-a3b0-c665420b70af 076decc351c94a3487594883f7420966 1b1e4fae15e045eaa66f8ee2c38e06ea - - -] Share type extra spec requirement "snapshot_support=<is> True" does not match reported capability "False" capabilities_satisfied /usr/lib/python3.6/site-packages/manila/scheduler/utils.py:176
2021-05-19 06:43:17.848 7 DEBUG manila.scheduler.filters.capabilities [req-296dc84b-c365-46d5-a3b0-c665420b70af 076decc351c94a3487594883f7420966 1b1e4fae15e045eaa66f8ee2c38e06ea - - -] host: 'hostgroup@cephfs#cephfs', free_capacity_gb: 888.86, pools: None fails resource_type extra_specs requirements host_passes /usr/lib/python3.6/site-packages/manila/scheduler/filters/capabilities.py:48
2021-05-19 06:43:17.848 7 INFO manila.scheduler.filters.base [req-296dc84b-c365-46d5-a3b0-c665420b70af 076decc351c94a3487594883f7420966 1b1e4fae15e045eaa66f8ee2c38e06ea - - -] Filter CapabilitiesFilter returned 0 host(s)
2021-05-19 06:43:17.848 7 ERROR manila.scheduler.manager [req-296dc84b-c365-46d5-a3b0-c665420b70af 076decc351c94a3487594883f7420966 1b1e4fae15e045eaa66f8ee2c38e06ea - - -] Failed to schedule create_share: No valid host was found. Failed to find a weighted host, the last executed filter was CapabilitiesFilter.: manila.exception.NoValidHost: No valid host was found. Failed to find a weighted host, the last executed filter was CapabilitiesFilter.
```
Running the following command fixed the problem for me:
```
openstack share type set cephfsnfstype --extra-specs snapshot_support=false
```